task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -504,6 +516,12 @@ def handle_requests(self, log: FilteringBoundLogger) -> Generator[None, bytes, N
             elif isinstance(msg, GetVariable):
                 var = self.client.variables.get(msg.key)
                 resp = var.model_dump_json(exclude_unset=True).encode()
+            elif isinstance(msg, PatchTIToDeferred):
+                self.final_state = IntermediateTIState.DEFERRED
+                self.client.task_instances.defer(self.ti_id, msg.model_dump_json())
+                # hmmm, can we do better here
+                # setting to "\n" as we do not have a response to return
```

Comment:
Needs some thought here @ashb @kaxil

Reply:
Two options:

1) Move `self.stdin.write(resp + b"\n")` to all the if's where we are sending a msg
2) Set `resp` to `None`. and change L529 to:

```python
if resp:
     self.stdin.write(resp + b"\n")
```

Reply:
Handled this using 2.